sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/v2/state/StateDataSourceTransformWithStateSuite.scala

```diff
@@ -1044,38 +1049,74 @@ class StateDataSourceTransformWithStateSuite extends StateStoreMetricsTest
             OutputMode.Append())
         testStream(query)(
           StartStream(checkpointLocation = tmpDir.getCanonicalPath),
-          AddData(inputData, (1, 1L), (2, 2L), (3, 3L)),
+          AddData(inputData, (1, 1L), (2, 2L), (3, 3L), (4, 4L)),
           ProcessAllAvailable(),
-          AddData(inputData, (2, 2L), (3, 3L), (4, 4L)),
+          AddData(inputData, (5, 1L), (6, 2L), (7, 3L), (8, 4L)),
           ProcessAllAvailable(),
-          AddData(inputData, (3, 3L), (4, 4L), (5, 5L)),
+          AddData(inputData, (9, 1L), (10, 2L), (11, 3L), (12, 4L)),
           ProcessAllAvailable(),
-          AddData(inputData, (4, 4L), (5, 5L), (6, 6L)),
+          AddData(inputData, (13, 1L), (14, 2L), (15, 3L), (16, 4L)),
           ProcessAllAvailable(),
-          AddData(inputData, (5, 5L), (6, 6L), (7, 7L)),
+          AddData(inputData, (17, 1L), (18, 2L), (19, 3L), (20, 4L)),
           ProcessAllAvailable(),
           // Ensure that we get a chance to upload created snapshots
           Execute { _ => Thread.sleep(5000) },
           StopStream
         )
       }
 
-      val stateSnapshotDf = spark
-        .read
-        .format("statestore")
-        .option("snapshotPartitionId", 0)
-        .option("snapshotStartBatchId", 1)
-        .option("stateVarName", "countState")
-        .load(tmpDir.getCanonicalPath)
-
-      val stateDf = spark
-        .read
-        .format("statestore")
-        .option("stateVarName", "countState")
-        .load(tmpDir.getCanonicalPath)
-        .filter(col("partition_id") === 0)
-
-      checkAnswer(stateSnapshotDf, stateDf)
+      // Create a file manager for the state store with opId=0 and partition=4
+      val dfsRootDir = new File(tmpDir.getAbsolutePath + "/state/0/4")
+      val fileManager = new RocksDBFileManager(
+        dfsRootDir.getAbsolutePath, Utils.createTempDir(), new Configuration,
+        CompressionCodec.LZ4)
+
+      // Read the changelog for one of the partitions
+      val changelogReader = fileManager.getChangelogReader(3, true)
+      val entries = changelogReader.toSeq
+      assert(entries.size == 2)
```

Comment:
Shall we add more info about version 3? Which keys are bound to which partition?

Reply:
Done